tests/agent/test_agent.py

```diff
@@ -142,3 +173,50 @@ def test_gx_agent_run_handles_subscriber_error_on_close(
     subscriber.close.side_effect = SubscriberError
     agent = GXAgent()
     agent.run()
+
+
+def test_gx_agent_updates_cloud_on_job_status(
+    subscriber, create_session, get_context, client, gx_agent_config, event_handler
+):
+    correlation_id = "4ae63677-4dd5-4fb0-b511-870e7a286e77"
+    url = f"{gx_agent_config.gx_cloud_base_url}/organizations/{gx_agent_config.gx_cloud_organization_id}/agent-jobs/{correlation_id}"
+    job_started_data = json.dumps(JobStarted().dict())
```

Comment:
```suggestion
    job_started_data = JobStarted().json()
```
We should use pydantic `.json()` output over `json.dumps()`.
It has awareness of many more types than the standard `json` packages.
https://docs.pydantic.dev/latest/usage/exporting_models/#modeljson